app/models/post.rb

```diff
@@ -1,2 +1,17 @@
 class Post < ApplicationRecord
+  has_many :likes, foreign_key: :post_id
+  has_many :comments, foreign_key: :post_id
+
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  after_save :update_post_counter
```

Comment:
- Kudos for using one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_counter 🔥🔥

Reply:
Thanks !